packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -1305,35 +1327,83 @@ class _CupertinoActionSheetActionState extends State<CupertinoActionSheetAction>
   }
 }
 
+// Base class for stateless widget classes that mark their [Element] as
+// [_SlideTarget].
+//
+// Using [_SlideTarget] requires injecting [_SlideTarget] objects as
+// [MetaData.data], which should be an object that is persistent across
+// rebuilds. Stateless widgets can use their elements as a persistent object.
+mixin _SlideTargetWidgetMixin on StatelessWidget {
+  // Called by the element when a pointer that is in touch with the screen moves
+  // from out of the widget to within.
+  void handleSlideIn();
+
+  // Called by the element when the user taps down or lifts up on the button.
+  //
+  // The boolean value is true if the user is tapping down on the button.
+  ValueSetter<bool>? get onPressStateChange;
+
+  @override
+  StatelessElement createElement() => _SlideTargetElement(this);
+}
+
+// A [Element] class that marks itself as `_SlideTarget`.
+class _SlideTargetElement extends StatelessElement implements _SlideTarget {
```

Comment:
This seems like unconventional in the framework. Have you consider creating a statefulWidget just for keep track of the state if we want to move the state out of `_ActionSheetButtonBackground`?

basically using composition than inheritance.

Reply:
This PR is to move the state out of `_ActionSheetButtonBackground`. I wanted to save a stateful widget since it really isn't needed. But yeah I can revert back to `StatefulWidget` with no state changes. I guess this shouldn't cause any performance difference.

Reply:
yeah I will still prefer an additional statefulwidget to manage the state since that is the pattern we use in framework. This won't be scalable for example if you have more than one mixin like this that try to override Element at the same time.